### PR TITLE
chore(test): replace babel-jest with @swc/jest (#4587)

### PR DESCRIPTION
**Pull request by msdlisper**
## Summary

Try To Improve Rspack Developer's Developer Experience #4587 

## Test Plan

cd to `packages/rspack` and  `pnpm run build && pnpm run test`

## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Comment by msdlisper** [comment]
@hardfist I tried it and it was OK, but it improved a lot, from around 160s to around 155s on my mac, but if there is a lot of `.tsx` testing behind it, the effect should show

**Comment by hardfist** [comment]
thanks！

**Comment by JSerFeng** [comment]
Thank you !

File: packages/rspack/tests/configCases/errors/loader-syntax-error/index.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 it("should report syntax error", () => {
-  let errored = false;
+	let errored = false;
 	try {
 		require("./lib");
 	} catch (e) {
-    errored = true;
-		expect(e.message).toContain("SyntaxError");
+		errored = true;
+		expect(e.message).toContain("Syntax Error");
```
Comment (hardfist) [inline packages/rspack/tests/configCases/errors/loader-syntax-error/index.js:7]: this changed because of swc？

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -1,5 +1,9 @@
 lockfileVersion: '6.0'
 
+settings:
```
Comment (hardfist) [inline pnpm-lock.yaml:3]: do you use corepack？seems your pnpm version is not the same with original？

File: packages/rspack/tests/configCases/errors/loader-syntax-error/index.js (hunk 1 above)
Comment (hardfist) [reply packages/rspack/tests/configCases/errors/loader-syntax-error/index.js:7]: @msdlisper can you help explain this change to me?
image

Comment (LingyuCoder) [reply packages/rspack/tests/configCases/errors/loader-syntax-error/index.js:7]: This difference is due to the different error messages between babel-jest and @swc/jest. So this change is safe

**Comment by LingyuCoder** [comment]
Thank you. It looks good to me. On my machine, it gives about 20% improvement (92s => 74s). Could you please resolve the conflicts with the main branch, and then I will merge this branch.

**Review by hardfist — APPROVED** [review]

**Pull request merged**